### ISSUE TITLE
Unable to reset tensorflow model

### ISSUE THREAD

**Issue by Tarungarg98** [tensorflow/tensorflow#36587]
I am trying to find the optimal number of neurons in the model. For that I am using a loop for number of layers. But instead of resetting the tensorflow model, it's adding layers to it. I tried K.clear_session(),sess.close(), gc.collect(). It doesn't works. Cuda.close() terminates the program, I am using tensorflow 1.10, win 10.

**Comment by Saduf2019** [comment]
@Tarungarg98 Could you please provide us with simple standalone code to reproduce the issue in our environment, Thanks

**Comment by Tarungarg98** [comment]
![image](https://user-images.githubusercontent.com/44539397/74142273-8d34d080-4c1e-11ea-9fa0-bb6a955c9ba9.png)


I am varying the number of neurons with variable y, but it's not creating a mew model with every loop

**Comment by Saduf2019** [comment]
@Tarungarg98  Please fill the template and provide complete editable code for us to replicate and help.

**Comment by Tarungarg98** [comment]
OS : Windows 10
Tensorflow version : 1.10.0 - gpu
Laptop: HP cc-102-tx
Cuda : 9.0
Graphics Card - NVIDIA GEForce 940MX 
RAM -8 GB

Code :
for x in ['relu','tanh']:
	for y in range(1,20):
		for z in [1500]:
			model = Sequential()
			model.add(Dense(y, input_dim=4, activation=x))
			model.add(Dense(2, activation='linear'))
			opt = tf.keras.optimizers.SGD(lr=0.01, momentum=0.9,decay=1e-5, nesterov=True)
			model.compile(loss='mean_squared_error', optimizer=opt,metrics=['accuracy'])
			model.fit(x_train, y_train, validation_data=(x_validate, y_validate), epochs=z,verbose=1,callbacks=[tensorboard_callback])

**Comment by Saduf2019** [comment]
@Tarungarg98  Could you please share complete implementable code, so we could replicate it in our environment to see where the issue lies and help you.

**Comment by Tarungarg98** [comment]
def nn(x,y,x_train,x_validate,x_test,y_train,y_validate,y_test,results,output):
	K.clear_session()
	model=None
	model= Sequential()
	model.add(Dense(y, input_dim=4, activation=x))
	model.add(Dense(2, activation='linear'))
	opt = tf.keras.optimizers.Adam(lr=0.01, decay=1e-3)
	model.compile(loss='mean_squared_error', optimizer=opt,metrics=['accuracy'])
	# fit model
	print('Training :',x,y)
	log_dir="logs/test3/" + x+'_'+str(y)+'_3000_'+datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
	tensorboard_callback = TensorBoard(log_dir=log_dir, histogram_freq=1)
	model.fit(x_train, y_train, validation_data=(x_validate, y_validate), epochs=3000,verbose=0,callbacks=[tensorboard_callback])

**Comment by Tarungarg98** [comment]
Every time I call this function I need to create a new model, but it's not creating one

**Comment by acvander** [comment]
I'm seeing a similar issue in my project using TF2.0. Should I open a separate issue referencing this one for that?

**Comment by Saduf2019** [comment]
@acvander Please create a separate issue for that.

**Comment by Saduf2019** [comment]
@Tarungarg98 I ahve tried to replicate issue faced by you with the code provided here is the [link](https://colab.sandbox.google.com/gist/Saduf2019/a538dda0514ef28bd545d4311d1c937e/untitled31.ipynb) for the same, unless we have the complete executable code[with dependencies] to see what issue is faced by you we will not be able to help.

**Comment by Tarungarg98** [comment]
I updated the code in the editor
It works on fine on colab but not on my PC
ON colab the model is resetting but on PC it's not resetting.

Attaching the snip of my results 
![image](https://user-images.githubusercontent.com/44539397/74315138-d874ed80-4d9c-11ea-9f9e-910bdf48ecd8.png)

**Comment by Tarungarg98** [comment]
[link](https://colab.research.google.com/gist/Tarungarg98/d7a247e1fe961f267f6c8d7cf6c7abeb/untitled31.ipynb)

**Comment by Saduf2019** [comment]
@Tarungarg98  Please share "data_nn.csv" to try this on local environment.

**Comment by Tarungarg98** [comment]
@Saduf2019  Added

**Comment by Saduf2019** [comment]
@Tarungarg98  I have not received the csv. please drag and drop the zip file in the comment box.

**Comment by Tarungarg98** [comment]
[data_nn.zip](https://github.com/tensorflow/tensorflow/files/4191100/data_nn.zip)

**Comment by Saduf2019** [comment]
@Tarungarg98 I have replicated your issue on my local as well [result is same as the colab link you have shared above] and understand that when you execute your code on colab the dependencies are taken care, whereas on local your imports were not as expected, ["keras.backend" from keras] please use:

from tensorflow.keras import backend as K instead of #import keras.backend as K

Could you please make this change and let us know if you still face the issue.

**Comment by Tarungarg98** [comment]
It works after making the change. Thank you
